kafka-client/src/main/java/dev/responsive/kafka/internal/db/mongo/MongoWindowedTable.java

```diff
@@ -483,7 +509,35 @@ public KeyValueIterator<WindowedKey, byte[]> fetchAll(
       final long timeFrom,
       final long timeTo
   ) {
-    throw new UnsupportedOperationException("fetchAll not yet supported for Mongo backends");
+    if (!timestampFirstOrder) {
+      throw new UnsupportedOperationException("Range queries such as fetchAll require stores to be "
```

Comment:
this is because mongo doesn't let you scan the whole table?

Reply:
I guess this was a "prevent users from shooting themselves in the foot" moment -- it's possible, just really inefficient since we have to scan everything in between (see comment about variable length keys).

Of course if we're going to go the route of scanning everything in order to enable partition scaling, then this constraint doesn't make any sense. I'll remove it